### PR TITLE
Codeine and heroin now can suppress cough

### LINKED ISSUE #16037: Codeine, heroin, tramadol don't treat cough
Contrary to codeine's own description.

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Closes #16037.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Did you test it? Because (at least) one of us misunderstood what does
```
"base_mods": { "cough_chance": [0] }
```
actually do.

The way I understand it, it's not a modifier but a chance of the effect to cause coughing.

**Comment by Night-Pryanik** [comment]
I looked at this text at `effects_json.md`: `"X_chance" - Basic chance of X triggering each time`
So I assumed that `"cough_chance": [0]` means 0 chance of triggering cough.

I test it for duration of 180 ticks that I set for the first time, then I changed it to 7200. Now I tested it longer, and the character still coughs during this period.

**Comment by Night-Pryanik** [comment]
Maybe then just increase cough chance from 1 in 300 caused by flu and cold to, say, 1 in 1000?

**Comment by Night-Pryanik** [comment]
Yeah, it seems I totally misunderstood the way it works. My PR just guarantees that taking codeine or heroine will not cause coughing _from these meds_ for 7200, but won't suppress cough from other origin.

**Comment by Coolthulhu** [comment]
Now it prevents coughing from smoke too. Should probably only prevent non-harmful coughing.

**Comment by Night-Pryanik** [comment]
If IIRC, these opiods just block the coughing center in the brain, so it will do prevent coughing of any origin.

**Comment by Coolthulhu** [comment]
But it should not prevent stamina and hp loss from smoke, just the waste of moves.

**Comment by mugling** [comment]
How does this relate to the existing cough syrup?

**Comment by Night-Pryanik** [comment]
It doesn't relate at all. Syrup works through resisting flu and common cold effects.
Cough syrup should probably be renamed to something like "anti-flu" syrup, because it not only block coughing, but also removing other negative effects from these diseases.

**Comment by Coolthulhu** [comment]
Now coughing only drains stamina when suppressed.

**Comment by Coolthulhu** [comment]
Also, without coughing, there is no feedback on why is the stamina/hp loss happening.

**Comment by Night-Pryanik** [comment]
> Now coughing only drains stamina when suppressed.

Is it? Inhaling smoke suppress cough, but penalties to stats are still applied.

> Also, without coughing, there is no feedback on why is the stamina/hp loss happening.

The game says about inhaling, I think it gives a hint on stamina/hp loss.

**Comment by Coolthulhu** [comment]
> Is it? Inhaling smoke suppress cough, but penalties to stats are still applied.

But it doesn't drain stamina and deal damage when not suppressed.
Read the code. When the player doesn't have the suppressant effect, the block with stamina drain and hp damage is skipped.

**Comment by Night-Pryanik** [comment]
Oh, now I got what you meant.

File: src/player.cpp [hunk 1]
```diff
@@ -5887,14 +5887,15 @@ void player::siphon( vehicle &veh, const itype_id &desired_liquid )
...
-        if( harmful ) {
-            const int stam = stamina;
-            mod_stat( "stamina", -100 );
-            if( stam < 100 && x_in_y( 100 - stam, 100 ) ) {
-                apply_damage( nullptr, bp_torso, 1 );
+    if( harmful ) {
+        const int stam = stamina;
+        mod_stat( "stamina", -100 );
+        if( stam < 100 && x_in_y( 100 - stam, 100 ) ) {
+            apply_damage( nullptr, bp_torso, 1 );
```
Comment (Coolthulhu) [inline src/player.cpp]: Indent

**Pull request merged**